// This file defines the specification for machines.

syntax = "proto2";

message MachineSpec {
  optional string name = 1;
  repeated Host host = 2;
  repeated MirroredDirectory mirrored_directory = 3;
}

message Host {
  optional string host = 1;
  optional int32 port = 2;
}

// A mirrored directory is a directory that is mirrored between a local and
// remote machine.
message MirroredDirectory {
  // An arbitrary name for the mirrored directory.
  optional string name = 1;
  // The absolute path of the directory as it exists on the "remote" machine.
  optional string remote_path = 2;
  // The absolute path of the directory as it exists on the local machine.
  optional string local_path = 3;
  // A list of exclude patterns passed to rsync when push or pulling changes
  // between local and remote machines.
  repeated string rsync_exclude = 4;

  // Set either of these flags to mark that the mirrored directory can only be
  // pushed or pulled to/from, respectively. Attempting to pull when push_only
  // or push when pull_only will error.
  optional bool push_only = 5;
  optional bool pull_only = 6;

  // Set this to the relative file of a path to use as a timestamp file.
  optional string timestamp_relpath = 7;

  // If set, disable error if the local_path does not exist.
  optional bool skip_if_not_present = 8 [default = false];
}
